app/src/main/java/net/squanchy/support/lang/Ids.java

```diff
@@ -0,0 +1,20 @@
+package net.squanchy.support.lang;
+
+import java.util.zip.CRC32;
+
+public final class Ids {
+
+    public static int safelyConvertIdToInt(String id) {
+        try {
+            return Integer.parseInt(id);
+        } catch (NumberFormatException e) {
+            return -1;
+        }
+    }
+
+    public static long checksumOf(String id) {
+        CRC32 checksum = new CRC32();
```

Comment:
Why do you create a new one every time?

In general also avoid util classes like these, you're already putting random stuff in here. Have a proper `Checksum` class which has a `CRC32 crc` field, and exposes a `checksum(String)` method. You can then only initialise it once and inject it with Dagger, or even just keep it around in whatever creates the models (it means you need to pass the `numericId` already calculated to `Event.create()`).